File: C#/booleanChecker.cs
using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var testRuns = 500;
        
        bool allTrue = true;
        var random = new Random();

        for (int i = 0; i < testRuns; i++)
        {
            if (!GenerateExpression(random))
            {
                allTrue = false;
                break;
            }
        }

        Console.WriteLine(allTrue);
    }

    static bool GetRandomBoolean(Random random)
    {
        return random.NextDouble() < 0.5;
    }

    static bool GenerateExpression(Random random)
    {
        bool[] x = Enumerable.Range(0, 5).Select(_ => GetRandomBoolean(random)).ToArray();

        bool ex1 = (x[0] && x[1] && x[2]) || (x[3] && x[4]);
        bool ex2 = x[0] && x[1] && x[2] || x[3] && x[4];

        return ex1 == ex2;
    }
}

File: C/booleanChecker.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define TEST_RUNS 500

int getRandomBoolean() {
    return rand() % 2;
}

int generateExpression() {
    int x[5];
    for (int i = 0; i < 5; i++) {
        x[i] = getRandomBoolean();
    }

    // Expression format 1
    int ex1 = ((x[0] && x[1] && x[2]) || (x[3] && x[4]));

    // Expression format 2
    int ex2 = (x[0] && x[1] && x[2] || x[3] && x[4]);

    return ex1 == ex2;
}

int main() {
    srand(time(NULL));

    int allTrue = 1;
    for (int i = 0; i < TEST_RUNS; i++) {
        if (!generateExpression()) {
            allTrue = 0;
            break;
        }
    }

    printf("%s\n", allTrue ? "true" : "false");

    return 0;
}
